internal/pkg/comm/creds.go

```diff
@@ -28,26 +29,50 @@ var (
 // NewServerTransportCredentials returns a new initialized
 // grpc/credentials.TransportCredentials
 func NewServerTransportCredentials(
-	serverConfig *tls.Config,
+	serverConfig *TLSConfig,
 	logger *flogging.FabricLogger) credentials.TransportCredentials {
-
 	// NOTE: unlike the default grpc/credentials implementation, we do not
 	// clone the tls.Config which allows us to update it dynamically
-	serverConfig.NextProtos = alpnProtoStr
+	serverConfig.lock.Lock()
+	defer serverConfig.lock.Unlock()
+	serverConfig.config.NextProtos = alpnProtoStr
 	// override TLS version and ensure it is 1.2
-	serverConfig.MinVersion = tls.VersionTLS12
-	serverConfig.MaxVersion = tls.VersionTLS12
+	serverConfig.config.MinVersion = tls.VersionTLS12
+	serverConfig.config.MaxVersion = tls.VersionTLS12
 	return &serverCreds{
 		serverConfig: serverConfig,
 		logger:       logger}
 }
 
 // serverCreds is an implementation of grpc/credentials.TransportCredentials.
 type serverCreds struct {
-	serverConfig *tls.Config
+	serverConfig *TLSConfig
 	logger       *flogging.FabricLogger
 }
 
+type TLSConfig struct {
+	config *tls.Config
+	lock   *sync.Mutex
```

Comment:
no need to have a pointer mutex. a non pointer is just fine, and saves you initialization.

Reply:
I'm not sure about this, w/o the pointer this triggers `copylocks` check failure with govet.

Reply:
But with your latest version of the code, you don't reference the lock externally anymore.

Reply:
I'm not sure what you mean by this statement, the `AddRootCA()` and `SetClientCAs()` still reference the TLConfig lock.

Reply:
Right, but these are all methods of the struct, so they reference it internally. 
By externally i mean that the lock is referenced by some other function which is not an instance method of the struct TLSConfig